pkg/store/bucket.go

```diff
@@ -1544,7 +1567,20 @@ func (r *bucketIndexReader) fetchPostings(groups []*postingGroup) error {
 				// Return postings and fill LRU cache.
 				// Truncate first 4 bytes which are length of posting.
 				groups[p.groupID].Fill(p.keyID, newBigEndianPostings(pBytes[4:]))
-				r.block.indexCache.StorePostings(r.ctx, r.block.meta.ULID, groups[p.groupID].keys[p.keyID], pBytes)
+
+				storeData := pBytes
+
+				if r.block.enablePostingsCompression {
+					// Reencode postings before storing to cache. If that fails, we store original bytes.
+					// This can only fail, if postings data was somehow corrupted,
+					// and there is nothing we can do about it. It's not worth reporting here.
+					data, err := diffVarintSnappyEncode(newBigEndianPostings(pBytes[4:]))
```

Comment:
Have you considered adding a couple of metrics with postings raw size and the compressed one, to measure the compression ratio in the wild?

Reply:
Good point

Reply:
I've added following metrics:
- number of compressions, and compression failures 
- total original size and compressed size
- total compression time

Reply:
Oh, one more thing... I've moved compression outside of the lock. However, storing data back to cache is still done with lock held, which looks wrong. I haven't fixed that part, because this PR ir not about it, but it looks like it should be moved outside as well. I haven't looked into more details here (like, does it break anything?)

Reply:
Also added number of decompressions, decompression errors, and total decompression time (snappy part only, we don't time decoding varints and adding values together)